src/commands/ext-dev-publish.ts

```diff
@@ -0,0 +1,36 @@
+import { Command } from "../command";
+import { publishExtensionVersionFromLocalSource } from "../extensions/extensionsHelper";
+import { findExtensionYaml } from "../extensions/localHelper";
+import { requireAuth } from "../requireAuth";
+import * as clc from "cli-color";
+import { FirebaseError } from "../error";
+
+/**
+ * Command for publishing an extension version.
+ */
+export default new Command("ext:dev:publish <publisherId>/<extensionId>")
+  .description(`publish a new version of an extension`)
+  .help(
+    "if you have not previously published a version of this extension, this will " +
+      "create the extension. If you have previously published a version of this extension, this version must " +
+      "be greater than previous versions."
+  )
+  .before(requireAuth)
+  .action(async (extensionRef: string, options: any) => {
+    const [publisherId, extensionId] = extensionRef.split("/");
+    if (!publisherId || !extensionId) {
```

Comment:
My understanding of Commander's behavior (that powers our CLI options) is that since you specify `<publisherId>/<extensionId>`, you'll need to have two parameters in the `action` function before `options` to accept both of those parameters. Can you verify this behavior for me?

Thinking it through a bit more, the CLI may actually be hiding such an error from you, in a way. I'd recommend you go in and check this out. (My gut tells me that `<publisherId>/<extensionId>` really needs to be `<extensionRef>` and it needs to be documented to say "in the form of 'publisherId/extensionId'" based on my past experience, but I may be wrong. I _think_ this is the only place we've had compound parameters like this. The fact that you're trying to split them makes me... curious.

Reply:
Good catch on this, you're right. the `/<extensionId>` part of the was never being parsed or used at all. 

Only one value was ever being passed in to the function, if I tried to make the signature
`.action(async (publisherId: string, extensionId: string, options: any) => {`
it would just pass in `options` value for `extensionId` field, and `options` field itself would just be null.